src/ExtEvLoop.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace React\EventLoop;
+
+use Ev;
+use EvLoop;
+use React\EventLoop\Tick\FutureTickQueue;
+use React\EventLoop\Timer\Timer;
+use SplObjectStorage;
+
+/**
+ * An `ext-ev` based event loop.
+ *
+ * This loop uses the [`ev` PECL extension](https://pecl.php.net/package/ev),
+ * that provides an interface to `libev` library.
+ *
+ * This loop is known to work with PHP 5.4 through PHP 7+.
+ *
+ * @see http://php.net/manual/en/book.ev.php
+ * @see https://bitbucket.org/osmanov/pecl-ev/overview
+ */
+class ExtEvLoop implements LoopInterface
+{
+    private $loop;
+    private $futureTickQueue;
+    private $timers;
+    private $readStreams = [];
+    private $writeStreams = [];
+    private $running;
+    private $signals;
+    private $signalEvents = [];
+
+    public function __construct()
+    {
+        $this->loop = new EvLoop();
+        $this->futureTickQueue = new FutureTickQueue();
+        $this->timers = new SplObjectStorage();
+        $this->signals = new SignalsHandler(
+            $this,
+            function ($signal) {
+                $this->signalEvents[$signal] = $this->loop->signal($signal, function () use ($signal) {
+                    $this->signals->call($signal);
+                });
+            },
+            function ($signal) {
+                if ($this->signals->count($signal) === 0) {
+                    unset($this->signalEvents[$signal]);
```

Comment:
This should probably `stop()` the watcher before clearing this reference, no?

Reply:
Yes, it's very important to stop watcher before clearing the reference. Fixed.